server/src/main/java/org/apache/iotdb/db/qp/utils/GroupByLevelController.java

```diff
@@ -136,6 +139,44 @@ public void control(ResultColumn rawColumn, List<ResultColumn> resultColumns)
     prevSize = resultColumns.size();
   }
 
+  // As one expression may have many aggregation results in the tree leaf, here we should traverse
+  // all the successor expressions and record the count(*) indices
+  private Set<Integer> getCountStarIndices(ResultColumn rawColumn) {
+    Set<Integer> countWildcardIterIndices = new HashSet<>();
+    int idx = 0;
+    for (Iterator<Expression> it = rawColumn.getExpression().iterator(); it.hasNext(); ) {
+      Expression expression = it.next();
+      if (expression instanceof FunctionExpression
+          && expression.isPlainAggregationFunctionExpression()
+          && ((FunctionExpression) expression).isCountStar()) {
+        countWildcardIterIndices.add(idx);
+      }
+      idx++;
+    }
+    return countWildcardIterIndices;
+  }
+
+  private void checkAliasAndUpdateAliasMap(ResultColumn rawColumn, String originName)
+      throws LogicalOptimizeException {
+    if (!rawColumn.hasAlias()) {
+      return;
+    } else if (aliasMap == null) {
+      aliasMap = new HashMap<>();
+    }
+    // If an alias is corresponding to more than one result column, throw an exception
+    if (aliasMap.get(originName) == null && aliasMap.containsValue(rawColumn.getAlias())) {
```

Comment:
Better to add a `Map<String, String> aliasToColumnName`, time complexity of `containsValue` in `HashMap` is O(N).

Reply:
I know that...But I did this based on that there are usually not so many aliases in one query, and there has been so many data structures that I don't want to add more.

Reply:
> there are usually not so many aliases in one query

Some users may not write the sql manual, they may concat the sql in their program, they may write hundreds or thousands of column in select clause instead of using `*`, like: `sum(s1) as s1_sum, sum(s2) as s2_sum, .... sum(s1000) as s1000_sum`. We won't know how users will write the sql. 

> there has been so many data structures

One more field won't be detrimental to the readability of the program if you add enough comments to explain the usage of that field you add.